egs/wsj/s5/steps/libs/nnet3/xconfig/gru.py

```diff
@@ -0,0 +1,414 @@
+        repair_nonlin_str = "self-repair-scale={0:.10f}".format(repair_nonlin) if repair_nonlin is not None else ''
+        affine_str = self.config['ng-affine-options']
+        # Natural gradient per element scale parameters
+        # TODO: decide if we want to keep exposing these options
+        ng_per_element_scale_options = self.config['ng-per-element-scale-options']
+        if re.search('param-mean', ng_per_element_scale_options) is None and \
+           re.search('param-stddev', ng_per_element_scale_options) is None:
+           ng_per_element_scale_options += " param-mean=0.0 param-stddev=1.0 "
+        pes_str = ng_per_element_scale_options
+
+        # formulation like:
+        # z = \sigmoid ( x_t * U^z + s_{t-1} * W^z ) // update gate
+        # r = \sigmoid ( x_t * U^r + s_{t-1} * W^r ) // reset gate
+        # h = \tanh ( x_t * U^h + s_{t-1} \dot r * W^h )
+        # s_t = ( 1 - z ) \dot h + z \dot s_{t-1}
+
+        # write bias and minus-scale
+        f = open(self.config['vars_path']+"/minus_one", 'w')
+        f.write(" [ ")
+        for i in range(cell_dim):
+            f.write("-1 ")
+        f.write("]\n")
+        f.close()
+
+        f = open(self.config['vars_path']+"/bias_one", 'w')
+        f.write(" [ ")
+        for i in range(cell_dim):
+            f.write("1 ")
+        f.write("]\n")
+        f.close()
+
+        configs = []
+        configs.append("# Update gate control : W_z* matrics")
+        configs.append("component name={0}.W_z.xs_z type=NaturalGradientAffineComponent input-dim={1} output-dim={2} {3}".format(name, input_dim + cell_dim, cell_dim, affine_str))
+        
+        configs.append("# Reset gate control : W_r* matrics")
+        configs.append("component name={0}.W_z.xs_r type=NaturalGradientAffineComponent input-dim={1} output-dim={2} {3}".format(name, input_dim + cell_dim, cell_dim, affine_str))
+
+        configs.append("# h related matrix : W_h* matrics")
+        configs.append("component name={0}.W_h.UW type=NaturalGradientAffineComponent input-dim={1} output-dim={2} {3}".format(name, input_dim + cell_dim, cell_dim , affine_str))
+        
+        configs.append("# Defining the non-linearities")
+        configs.append("component name={0}.z type=SigmoidComponent dim={1} {2}".format(name, cell_dim, repair_nonlin_str))
+        configs.append("component name={0}.r type=SigmoidComponent dim={1} {2}".format(name, cell_dim, repair_nonlin_str))
+        configs.append("component name={0}.h type=TanhComponent dim={1} {2}".format(name, cell_dim, repair_nonlin_str))
+
+        configs.append("# Defining the components for other cell computations")
+        configs.append("component name={0}.h1 type=ElementwiseProductComponent input-dim={1} output-dim={2}".format(name, 2 * cell_dim, cell_dim))
+        configs.append("component name={0}.y1 type=ElementwiseProductComponent input-dim={1} output-dim={2}".format(name, 2 * cell_dim, cell_dim))
```

Comment:
When do you plan to add the 'fast' versions of these layers, that use the compoennts I added?